test/core/libp2p/multi/multistream_test.cpp

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <spdlog/spdlog.h>
+#include <gtest/gtest.h>
+
+#include "libp2p/multi/multistream.hpp"
+
+using libp2p::multi::Multistream;
+using kagome::common::Buffer;
+using kagome::expected::Value;
+using kagome::expected::Error;
+
+/**
+ * @given a protocol description and a buffer with data
+ * @when creating a multistream object
+ * @then a multistream object containing the protocol info and the data is created
+ */
+TEST(Multistream, Create) {
+  Multistream m =
+      Multistream::create("/bittorrent.org/1.0", Buffer({1, 2, 3, 4})).match(
+          [](Value<Multistream>& v) -> Multistream { return v.value;},
+          [](Error<std::string>& e) -> Multistream { ADD_FAILURE() << e.error;});
+
+  spdlog::info(m.getBuffer().toHex());
+  ASSERT_EQ(m.getCodecPath(), "/bittorrent.org/1.0");
+  ASSERT_EQ(Buffer({1, 2, 3, 4}), m.getEncodedData());
+
+  Multistream m1 = m;
+
+  ASSERT_EQ(m1.getCodecPath(), "/bittorrent.org/1.0");
+  ASSERT_EQ(Buffer({1, 2, 3, 4}), m.getEncodedData());
+}
+
+/**
+ * @given a buffer with bytes, which is a valid representation of a multistream
+ * @when creating a multistream object
+ * @then a multistream object is created from data in the buffer
+ */
+TEST(Multistream, CreateFromBuffer) {
+  Buffer buf;
+  buf.putUint8(0x05);
+  buf.put("/ipfs/Qmaa4Rw81a3a1VEx4LxB7HADUAXvZFhCoRdBzsMZyZmqHD/ipfs.protocol");
+  buf.putUint8('\n');
+  buf.put({1, 2, 3, 4, 5});
+  Multistream m2 = Multistream::createFromBuffer(buf).match(
+      [](Value<Multistream>& v) -> Multistream { return v.value;},
+      [](Error<std::string>& e) -> Multistream { ADD_FAILURE() << e.error;});
+
+  ASSERT_EQ(m2.getCodecPath(), "/ipfs/Qmaa4Rw81a3a1VEx4LxB7HADUAXvZFhCoRdBzsMZyZmqHD/ipfs.protocol");
+  ASSERT_EQ(Buffer({1, 2, 3, 4, 5}), m2.getEncodedData());
+}
```

Comment:
When you add more methods, please don't forget to test them as well

Reply:
What do you mean? Are there not tested methods?

Reply:
No, I meant when you add `rmPrefix(..)` etc